packages/ui/src/2_molecules/TransactionID/TransactionID.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useMemo } from 'react';
+
+import classNames from 'classnames';
+
+import { Icon } from '../../1_atoms';
+import { COPY, NEW_TAB } from '../../1_atoms/Icon/iconNames';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { TooltipPlacement, TooltipTrigger } from '../Tooltip/Tooltip.types';
+import styles from './TransactionID.module.css';
+import { blockExplorers, ChainId, currentChainId } from './TransactionID.types';
+
+interface TransactionIDProps {
+  value: string;
+  startLength?: number;
+  endLength?: number;
+  className?: string;
+  hideTooltip?: boolean;
+  chainId?: number | ChainId;
+  dataLayoutId?: string;
+  isNativeBtc?: boolean;
+  isAddress?: boolean;
+}
+
+export const TransactionID: React.FC<TransactionIDProps> = ({
+  startLength = 6,
+  endLength = 4,
+  className,
+  hideTooltip,
+  chainId,
+  dataLayoutId,
+  isNativeBtc,
+  isAddress,
+  value,
+}) => {
+  const transactionId = useMemo(() => {
+    if (value?.length && startLength && endLength) {
+      const start = value.substr(0, startLength);
+      const end = value.substr(-endLength);
+      return `${start} ··· ${end}`;
+    }
+    return value;
+  }, [value, startLength, endLength]);
+
+  const url = useMemo(() => {
+    if (isNativeBtc) {
+      return blockExplorers[`btc_${currentChainId}`];
+    }
+    return blockExplorers[chainId || currentChainId];
+  }, [isNativeBtc, chainId]);
+
+  const copyAddress = useCallback(async () => {
+    await navigator.clipboard.writeText(value);
+    alert('Address was copied to clipboard.');
+  }, [value]);
```

Comment:
```suggestion
  const onCopyAddress = useCallback(
    () => navigator.clipboard.writeText(value),
    [value],
  );
```
I am not sure if we want to have `alert` after copy or not, pls discuss it with Gilad.

Reply:
this can be left in for now, we'll be added a Toast component soon to handle notifications like this